core/tools/generate_names.cc

```diff
@@ -486,15 +487,15 @@ void emit_register(ostream &out) {
     }
     out << '\n';
     for (auto &name : names) {
-        out << "    ENFORCE(" << name.srcName << "_id._id == " << name.id << "); /* " << name.srcName << "() */"
-            << '\n';
+        out << "    ENFORCE(" << name.srcName << "_id." << (name.isConstant ? "constantIndex" : "utf8Index")
+            << "() == " << name.id << "); /* " << name.srcName << "() */" << '\n';
     }
     out << '\n';
     out << "}" << '\n';
 }
 
 int main(int argc, char **argv) {
-    int i = 1;
+    int i = 0;
```

Comment:
This is OK now because the tag bits differentiate the valid name from the `noName` case, right?

Reply:
Yup.